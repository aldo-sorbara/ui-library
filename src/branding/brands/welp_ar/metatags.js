export const presto = {
  title: 'Presto Hoy Prestamos Online',
  description: 'Préstamo confiable, rápido y seguro',
  keywords: 'Préstamo, Crédito, Minicrédito, Presto Hoy, Efectivo, Cash'
};
export const posta = {
  title: 'Créditos Posta',
  description: 'Créditos Posta',
  keywords: 'Préstamo, Crédito, Posta, Efectivo, Cash'
};
export const welp = {
  title: 'W E L P',
  description: 'WELP',
  keywords: 'Préstamo, Crédito, Welp, Efectivo, Cash'
};
export default {
  title: 'Onboarding Title',
  description: 'Onboarding Description',
  keywords: 'Onboarding, React, Redux, Styled'
};
