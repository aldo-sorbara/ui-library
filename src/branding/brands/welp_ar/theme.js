import alerts from './jsons/alerts.json';
import assetsPaths from './jsons/assets.json';
import breakPoints from '../common/jsons/breakPoints.json';
import font from './jsons/font.json';
import palette from './jsons/palette.json';
import reset from './jsons/reset.json';
import spacer from './jsons/spacer.json';
import typography from './jsons/typography.json';
import zIndex from '../common/jsons/zIndex.json';

import buttons from './jsons/buttons.json';
import radioButton from './jsons/radioButton.json';
import list from './jsons/list.json';
import containers from './jsons/containers.json';
import modals from './jsons/modals.json';

export default {
  name: 'welp_ar',
  alerts,
  assetsPaths,
  breakPoints,
  buttons,
  list,
  containers,
  emSize: '16px',
  font,
  modals,
  palette,
  radioButton,
  reset,
  spacer,
  typography,
  zIndex
};
