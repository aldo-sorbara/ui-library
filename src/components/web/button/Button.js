import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * @render react
 * @name Button
 * @description Button Component for Wenance.
 * @example
 * <ButtonContainer>
 *   <Button data="button"> Button</Button>
 * </ButtonContainer>
 */

export const Button = props => {
  const { children, data, href } = props;
  const Component = href ? StyledLink : StyledButton;
  return (
    <Component {...props} data-w={data}>
      {children}
    </Component>
  );
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  data: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  isBlocked: PropTypes.bool,
  modifiers: PropTypes.arrayOf(PropTypes.string)
};

Button.defaultProps = {
  disabled: false,
  isBlocked: false,
  modifiers: []
};

const CommonButton = props => `
  border-radius: ${props.theme.buttons.borderRadius};
  display: ${props.isBlocked ? 'block' : 'inline-block'};
  cursor: pointer;
  font-family: inherit;
  opacity: ${props.disabled ? '0.3' : '1'};
  padding: ${props.theme.spacer.medium} ${props.theme.spacer.wrapperInner};
  text-align: center;
  transition: background-color 0.4s ease-in-out;
  touch-action: manipulation;
  user-select: none;
  vertical-align: middle;
  white-space: nowrap;
  width: ${props.isBlocked ? '100%' : 'auto'};
  cursor: ${props.disabled ? 'not-allowed' : 'pointer'};
  background-color: ${props.theme.palette.primary.default};
  border: 1px solid transparent;
  color: ${props.theme.palette.common.text.inverse};

  &:hover:enabled {
    background-color: ${props.theme.palette.primary.dark};
  }

  &:focus {
    outline: none;
  }

  &:disabled,
  button[disabled] {
    opacity: 0.3;
  }
`;

CommonButton.defaultProps = {
  theme: {
    palette: {
      common: {
        text: {
          inverse: '#ffffff'
        }
      },
      primary: {
        default: '#0082ff',
        dark: '#3a66ab'
      }
    },
    typography: {
      button: {
        fontSize: '.875rem',
        fontWeight: '700',
        lineHeight: '1.145'
      }
    },
    spacer: {
      medium: '1rem',
      large: '1.5rem',
      button: {
        smallSpacer: '.75rem'
      }
    },
    buttons: {
      borderRadius: '0'
    }
  }
};

const StyledButton = styled.button`
  ${CommonButton}
  ${props => props.theme.typography.button};
  ${props => props.modifiers.map(modifier => modifiers[modifier])}
`;

const StyledLinkButton = props => `
  background: transparent;
  border: 1px solid transparent;
  color: ${props.theme.palette.primary.default};
  &:hover, &:hover:enabled {
    background-color: inherit;
    opacity: .65
  }`;

const StyledLink = styled.a`
  ${CommonButton}
  ${props => props.theme.typography.button};
  ${props => props.modifiers.map(modifier => modifiers[modifier])}
  text-decoration: none;
`;

const StyledSmallButton = props => `
  padding: ${props.theme.spacer.button.smallSpacer} ${props.theme.spacer.large};
`;

const StyledInverseButton = props => `
  background: transparent;
  color: ${props.theme.palette.common.text.inverse};
  border: 1px solid ${props.theme.palette.common.text.inverse};
  
  &:hover, &:hover:enabled {
    background-color: inherit;
    opacity: .65
  }
`;

const StyledRoundedButton = () => `
  border-radius: 50%;
  width: 50px;
  height: 50px
  padding: 0;

  svg {
    display: block;
    margin: auto;
  }
`;

const modifiers = {
  small: StyledSmallButton,
  rounded: StyledRoundedButton,
  link: StyledLinkButton,
  inverse: StyledInverseButton
};

// ButtonContainer Component
export const ButtonContainer = props => {
  return (
    <StyledButtonContainer {...props}>
      <div>{props.children}</div>
    </StyledButtonContainer>
  );
};

const StyledButtonContainer = styled.div`
  padding: ${props => props.theme.buttons.padding};
`;

StyledButtonContainer.defaultProps = {
  theme: {
    buttons: {
      padding: '0 2rem 2rem'
    }
  }
};
