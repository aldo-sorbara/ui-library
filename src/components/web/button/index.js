export { Button, ButtonContainer } from './Button';
