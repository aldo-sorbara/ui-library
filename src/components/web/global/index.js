export {
  Container,
  Wrapper,
  Headline,
  Description,
  Legales,
  Label,
  LabelLarge,
  SecondHeadline,
  ContainerDesktop,
  Desktop,
  Onboarding,
  Terms,
  H1,
  SectionDescription,
  UrlLink,
  Timer,
  DesktopText,
  Bind,
  Logo,
  LogoMobile,
  TextPromotion
} from './Global';
