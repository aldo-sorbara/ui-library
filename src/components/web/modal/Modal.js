import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Transition } from 'react-spring/renderprops.cjs.js';

/**
 * @render react
 * @name Modal
 * @description Card Component for Wenance
 * @example
 * <Modal> This is a modal</Modal>
 */

export const Modal = props => {
  const {
    children,
    title,
    open,
    onClose,
    actionButtons,
    style,
    customStyle,
    closeButton,
    ...rest
  } = props;
  const [windowWidth, setWindowWidth] = useState(document.body.clientWidth);
  const handleResize = e => {
    setWindowWidth(document.body.clientWidth);
  };
  useEffect(() => {
    window.addEventListener('resize', handleResize, true);
    return () => {
      window.removeEventListener('resize', handleResize, true);
    };
  });
  return (
    <Transition
      items={open}
      from={{
        opacity: 0,
        transform: 'translateY(300%)'
      }}
      enter={{
        opacity: 1,
        transform: 'translateY(0%)'
      }}
      leave={{
        opacity: 0,
        transform: 'translateY(300%)'
      }}
    >
      {item =>
        item &&
        (p => (
          <BgModal {...rest} style={{ ...style, opacity: p.opacity }}>
            <div
              style={{
                width:
                  /* istanbul ignore next */ windowWidth > 1000 ? '100%' : 0
              }}
            />
            <ModalContainer windowWidth={windowWidth}>
              <StyledModal style={customStyle}>
                {onClose && !closeButton ? (
                  <Close onClick={onClose}>
                    <CloseIcon />
                  </Close>
                ) : (
                  <Close onClick={onClose}>{closeButton}</Close>
                )}
                {title && <Title>{title}</Title>}

                <div style={{ transform: p.transform }}>{children}</div>
                {actionButtons && (
                  <ActionContainer>{actionButtons}</ActionContainer>
                )}
              </StyledModal>
            </ModalContainer>
          </BgModal>
        ))
      }
    </Transition>
  );
};

Modal.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
    PropTypes.array
  ]),
  open: PropTypes.bool,
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  onClose: PropTypes.func,
  actionButtons: PropTypes.node,
  closeButton: PropTypes.node,
  windowWidth: PropTypes.number
};

Modal.defaultProps = {
  children: null,
  title: null,
  onClose: null,
  actionButtons: null,
  closeButton: null,
  windowWidth: 0,
  customStyle: {},
  style: {},
  open: false
};

const CloseIcon = props => (
  <svg
    xmlns='http://www.w3.org/2000/svg'
    fill='currentColor'
    height='24'
    width='24'
    viewBox='0 0 35 35'
  >
    {/* eslint-disable max-len */}
    <path d='M26 22.87A2.14 2.14 0 0 1 26 26a2.14 2.14 0 0 1-3.13 0l-5.54-5.54-5.5 5.5c-.45.45-.98.68-1.58.7A2.1 2.1 0 0 1 8.67 26 2.1 2.1 0 0 1 8 24.42c.01-.6.24-1.13.7-1.58l5.5-5.5-5.54-5.55a2.15 2.15 0 0 1 0-3.12 2.14 2.14 0 0 1 3.12 0l5.54 5.54 5.54-5.54a2.15 2.15 0 0 1 3.13 0 2.14 2.14 0 0 1 0 3.12l-5.54 5.54L26 22.87z' />
    {/* eslint-enable max-len */}
  </svg>
);

const BgModal = styled.div`
  display: flex;
  height: 100%;
  position: fixed;
  right: 0;
  top: 0;
  width: 100%;
  z-index: ${props => props.theme.zIndex.modal};
`;

const ModalContainer = styled.div`
  background-color: ${props => props.theme.modals.backgroundOverlay};
  display: flex;
  height: 100%;
  overflow: auto;
  width: ${/* istanbul ignore next */ props =>
    props.windowWidth >= 1000 ? '768px' : '100%'};
`;

const StyledModal = styled.div`
  background-color: ${props =>
    props.theme.modals.backgroundModal || props.theme.palette.primary.light};
  color: ${props => props.theme.palette.common.text.inverse};
  flex: 1;

  margin-left: auto;
  margin-right: auto;
  overflow: auto;
  padding: ${props => props.theme.spacer.large};
  position: relative;

  > div {
    margin-bottom: ${props => props.theme.spacer.large};
  }

  @media (min-width: ${props => props.theme.breakPoints.sm}) {
    border-radius: ${props => props.theme.modals.borderRadius};
    margin: ${props => props.theme.spacer.large};
    padding: ${props => props.theme.spacer.wrapperInner};
  }
`;

const ActionContainer = styled.div`
  display: flex;
  justify-content: space-between;
`;

const Close = styled.div`
  cursor: pointer;
  display: inline-block;
  position: absolute;
  right: ${props => props.theme.spacer.large};
  top: ${props => props.theme.spacer.large};

  svg {
    height: 2rem;
    width: 2rem;
  }
`;

const Title = styled.h1`
  margin-bottom: ${props => props.theme.spacer.blockSpacer};
  margin-top: 2.5rem;
  ${props => props.theme.typography.headline};
`;

export default Modal;
