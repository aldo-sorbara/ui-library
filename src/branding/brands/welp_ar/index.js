import wording from './jsons/wording.json';
import theme from './theme';
import config from './jsons/config.json';
import feedback from './jsons/feedback.json';
import assets from './jsons/assets.json';
import palette from './jsons/palette.json';

export default {
  wording: { onboarding: wording },
  feedback: { onboarding: feedback },
  theme,
  config: { onboarding: config },
  typography: theme.font.fontfamily,
  assets: { onboarding: assets },
  palette
};
