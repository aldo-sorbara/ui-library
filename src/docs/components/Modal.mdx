---
name: Modal
menu: Components
route: /modal
---

import { Playground, PropsTable } from 'docz';
import { Fragment } from 'react';
import Modal from '../../components/web/modal';
import { Modal as ModalProps } from '../../components/web/modal/Modal';
import { ThemeProvider } from 'styled-components';
import welp_ar from '../../branding/brands/welp_ar';
import GlobalStyle from '../../components/web/globalstyle';
import { Button, ButtonContainer } from '../../components/web/button';
import { styles } from '../../../utils/docz';

# Modal

<PropsTable of={ModalProps} />

## Basic usage

<Playground>
  <ThemeProvider theme={welp_ar.theme}>
    <>
      <GlobalStyle theme={welp_ar.theme} />
      <Modal
        className='position-unset'
        onClose={() => {
          console.log('closed');
        }}
        open
        closeButton={
          <Button
            onClick={() => console.log('Click')}
            data='button'
            modifiers={['rounded']}
          >
            <svg
              xmlns='http://www.w3.org/2000/svg'
              fill='currentColor'
              height='24'
              width='24'
              viewBox='0 0 35 35'
            >
              {/* eslint-disable max-len */}
              <path d='M26 22.87A2.14 2.14 0 0 1 26 26a2.14 2.14 0 0 1-3.13 0l-5.54-5.54-5.5 5.5c-.45.45-.98.68-1.58.7A2.1 2.1 0 0 1 8.67 26 2.1 2.1 0 0 1 8 24.42c.01-.6.24-1.13.7-1.58l5.5-5.5-5.54-5.55a2.15 2.15 0 0 1 0-3.12 2.14 2.14 0 0 1 3.12 0l5.54 5.54 5.54-5.54a2.15 2.15 0 0 1 3.13 0 2.14 2.14 0 0 1 0 3.12l-5.54 5.54L26 22.87z' />
              {/* eslint-enable max-len */}
            </svg>
          </Button>
        }
        title='This is a title'
        actionButtons={
          <Button isBlocked data='got-it'>
            Got it!
          </Button>
        }
      >
        This is the content.
      </Modal>
    </>
  </ThemeProvider>
</Playground>

[Try it on Storybook!](./storybook/index.html?path=/story/modal--with-title)
