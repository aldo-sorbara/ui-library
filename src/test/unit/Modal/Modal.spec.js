import React from 'react';
import { Button } from '../../../components/web/button';
import Modal from '../../../components/web/modal';
import renderer from 'react-test-renderer';
import { mountWithTheme } from '../../../../utils/helper';
import welp_ar from '../../../branding/brands/welp_ar';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

describe('Modal component', () => {
  it('should render snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Modal open className='test-card'>
          <div>test</div>
        </Modal>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render a modal with title', () => {
    const wrapper = mountWithTheme(
      <Modal open className='test-card' title='title'>
        <div>test</div>
      </Modal>,
      welp_ar.theme
    );
    expect(wrapper.props().title).toBe('title');
    expect(wrapper.find('h1').at(0)).not.toBe(undefined);
  });
  it('should render a modal with close button', () => {
    const onClose = jest.fn();
    const wrapper = mountWithTheme(
      <Modal open className='test-card' title='title' onClose={onClose}>
        <div>test</div>
      </Modal>,
      welp_ar.theme
    );
    expect(wrapper.props().onClose).not.toBe(null);
    wrapper
      .find('div')
      .at(4)
      .simulate('click');
    expect(onClose).toHaveBeenCalled();
  });
  it('should render a modal with closeButton prop', () => {
    const onClose = jest.fn();
    const wrapper = mountWithTheme(
      <Modal
        open
        className='test-card'
        title='title'
        closeButton={
          <Button data='button' modifiers={['rounded']}>
            <svg
              xmlns='http://www.w3.org/2000/svg'
              fill='currentColor'
              height='24'
              width='24'
              viewBox='0 0 35 35'
            >
              {/* eslint-disable max-len */}
              <path d='M26 22.87A2.14 2.14 0 0 1 26 26a2.14 2.14 0 0 1-3.13 0l-5.54-5.54-5.5 5.5c-.45.45-.98.68-1.58.7A2.1 2.1 0 0 1 8.67 26 2.1 2.1 0 0 1 8 24.42c.01-.6.24-1.13.7-1.58l5.5-5.5-5.54-5.55a2.15 2.15 0 0 1 0-3.12 2.14 2.14 0 0 1 3.12 0l5.54 5.54 5.54-5.54a2.15 2.15 0 0 1 3.13 0 2.14 2.14 0 0 1 0 3.12l-5.54 5.54L26 22.87z' />
              {/* eslint-enable max-len */}
            </svg>
          </Button>
        }
        onClose={onClose}
      >
        <div>test</div>
      </Modal>,
      welp_ar.theme
    );
    expect(wrapper.props().onClose).not.toBe(null);
    wrapper
      .find('div')
      .at(4)
      .simulate('click');
    expect(onClose).toHaveBeenCalled();
    expect(wrapper.props().closeButton).not.toBe(null);
  });
  it('should render a modal with action buttons', () => {
    const wrapper = mountWithTheme(
      <Modal open className='test-card' actionButtons={<div>test</div>}>
        <div>test</div>
      </Modal>,
      welp_ar.theme
    );
    expect(wrapper.props().actionButtons).not.toBe(undefined);
  });
  it('should render a modal with windowWidth > 1000', () => {
    const wrapper = mountWithTheme(
      <Modal open className='test-card' windowWidth={1050}>
        <div>test</div>
      </Modal>,
      welp_ar.theme
    );
    expect(wrapper.props().windowWidth).toBe(1050);
    window.dispatchEvent(new Event('resize'));
    wrapper.unmount();
  });
});
