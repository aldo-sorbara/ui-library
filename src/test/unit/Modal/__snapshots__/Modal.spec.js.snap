// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should render snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  position: fixed;
  right: 0;
  top: 0;
  width: 100%;
  z-index: 1050;
}

.c1 {
  background-color: rgba(255,255,255,0.75);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  overflow: auto;
  width: 100%;
}

.c2 {
  background-color: #73baff;
  color: #ffffff;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-left: auto;
  margin-right: auto;
  overflow: auto;
  padding: 1.5rem;
  position: relative;
}

.c2 > div {
  margin-bottom: 1.5rem;
}

.c3 {
  cursor: pointer;
  display: inline-block;
  position: absolute;
  right: 1.5rem;
  top: 1.5rem;
}

.c3 svg {
  height: 2rem;
  width: 2rem;
}

@media (min-width:768px) {
  .c2 {
    margin: 1.5rem;
    padding: 2rem;
  }
}

<div
  className="test-card c0"
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <div
    style={
      Object {
        "width": 0,
      }
    }
  />
  <div
    className="c1"
  >
    <div
      className="c2"
      style={Object {}}
    >
      <div
        className="c3"
        onClick={null}
      />
      <div
        style={
          Object {
            "transform": "translateY(300%)",
          }
        }
      >
        <div>
          test
        </div>
      </div>
    </div>
  </div>
</div>
`;
